lib/rust/profiler/src/lib.rs

```diff
@@ -0,0 +1,333 @@
+use std::cell;
```

Comment:
No module doc – this doc is super important here. This is the root of the library and this doc should be something a new team member or a new user of this library will read to understand how the whole library works. It should be a **complete** introduction to the profiler framework.

This doc should be a lengthy one – it should provide the user with as much information as possible. In particular, it should contain info about motivation, usage patterns, performance, disabling and enabling profilers, the guidance (when to use which profiler), etc. In fact, the doc here should be a shorter version of the design document and it should contain sections and should be nicely layouted.